src/plugins/controls/public/control_group/editor/control_group_editor.tsx

```diff
@@ -0,0 +1,153 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiFlyoutHeader,
+  EuiButtonGroup,
+  EuiFlyoutBody,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiFlyoutFooter,
+  EuiButton,
+  EuiFormRow,
+  EuiButtonEmpty,
+  EuiSpacer,
+  EuiCheckbox,
+} from '@elastic/eui';
+
+import { ControlGroupStrings } from '../control_group_strings';
+import { ControlStyle, ControlWidth } from '../../types';
+import { CONTROL_LAYOUT_OPTIONS, CONTROL_WIDTH_OPTIONS } from './editor_constants';
+
+interface EditControlGroupProps {
+  width: ControlWidth;
+  controlStyle: ControlStyle;
+  setAllWidths: boolean;
+  updateControlStyle: (controlStyle: ControlStyle) => void;
+  updateWidth: (newWidth: ControlWidth) => void;
+  updateAllControlWidths: (newWidth: ControlWidth) => void;
+  onCancel: () => void;
+  onClose: () => void;
+}
+
+export const ControlGroupEditor = ({
+  width,
+  controlStyle,
+  setAllWidths,
+  updateControlStyle,
+  updateWidth,
+  updateAllControlWidths,
+  onCancel,
+  onClose,
+}: EditControlGroupProps) => {
+  const [currentControlStyle, setCurrentControlStyle] = useState(controlStyle);
+  const [currentWidth, setCurrentWidth] = useState(width);
+  const [applyToAll, setApplyToAll] = useState(setAllWidths);
+
+  return (
+    <>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle size="m">
+          <h2>{ControlGroupStrings.management.getFlyoutTitle()}</h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiFormRow label={ControlGroupStrings.management.getLayoutTitle()}>
+          <EuiButtonGroup
+            color="primary"
+            idSelected={currentControlStyle}
+            legend={ControlGroupStrings.management.controlStyle.getDesignSwitchLegend()}
+            options={CONTROL_LAYOUT_OPTIONS}
+            onChange={(newControlStyle: string) => {
+              setCurrentControlStyle(newControlStyle as ControlStyle);
+            }}
+          />
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        <EuiFormRow label={ControlGroupStrings.management.getDefaultWidthTitle()}>
+          <EuiButtonGroup
+            color="primary"
+            idSelected={currentWidth}
+            legend={ControlGroupStrings.management.controlWidth.getWidthSwitchLegend()}
+            options={CONTROL_WIDTH_OPTIONS}
+            onChange={(newWidth: string) => {
+              setCurrentWidth(newWidth as ControlWidth);
+            }}
+          />
+        </EuiFormRow>
+        <EuiSpacer size="s" />
+        <EuiCheckbox
```

Comment:
Yup! I figured since I was able to get it working, I might as well keep it for now :) Definitely interested in returning to whether or not it's worth keeping, but figured it belonged in a separate PR regardless!